File: favourite-lyrics-domain/src/main/java/org/jesperancinha/lyrics/domain/port/LyricsPersistencePort.java
package org.jesperancinha.lyrics.domain.port;

import org.jesperancinha.lyrics.domain.data.LyricsDto;
import org.jesperancinha.lyrics.domain.data.LyricsFullDto;

import java.util.List;
import java.util.UUID;

public interface LyricsPersistencePort {

    void addLyrics(LyricsDto lyricsDto);

    void removeLyrics(LyricsDto lyricsDto);

    void updateLyrics(LyricsDto lyricsDto);

    List<LyricsDto> getAllLyrics();

    List<LyricsFullDto> getAllLFullLyrics();

    LyricsDto getLyricsById(UUID lyricsId);
}

File: favourite-lyrics-domain/src/main/kotlin/org/jesperancinha/lyrics/domain/port/LyricsPersistencePort.kt
package org.jesperancinha.lyrics.domain.port

import org.jesperancinha.lyrics.domain.data.LyricsDto
import org.jesperancinha.lyrics.domain.data.LyricsFullDto
import java.util.*

interface LyricsPersistencePort {
    fun addLyrics(lyricsDto: LyricsDto?)
    fun removeLyrics(lyricsDto: LyricsDto?)
    fun updateLyrics(lyricsDto: LyricsDto?)
    @JvmField
    val allLyrics: List<LyricsDto?>?
    val allLFullLyrics: List<LyricsFullDto?>?
    fun getLyricsById(lyricsId: UUID?): LyricsDto?
}